### PR TITLE
Ft/users

### PR DESCRIPTION
**Pull request by MainaMary**
-The userValidation to check if a user exists with the same username is not throwing the error message as expected

### CODE REVIEW

File: src/middleware/userValidator.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import UserModel from "../models/user"
+
+const userModel = new UserModel();
+class userValidator {
+    validateUserData(req, res, next){
+        const {email, password, userName} = req.body;
+        if(!email){
```
Comment (MbugwaSami) [inline src/middleware/userValidator.js]: Trim for empty spaces

File: src/models/user.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  getAllUsers(){
+    return allUsers
+  }
+
+  getSingleUser(reqData){
+    const {key, value}= reqData;
+    const user = allUsers.find(user=>{user[key]=== value})
+    return user
+  }
+  updateSingleUser(reqData){
```
Comment (MbugwaSami) [inline src/models/user.js]: You not updating the user data

File: src/models/user.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    const {key, value} = reqData
+  const singleUser = allUsers.find(user => user.key === value )
+  return singleUser
+  }
+
+
+  deleteUser(id){
+   const user= allUsers.filter(id)
```
Comment (MbugwaSami) [inline src/models/user.js]: For delete you should just have allUsers.filter(user => user.id !== id), But do a validation in your middleware to check if it exists.

File: src/modules/users/userControllers.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        const userDelete =deleteUser(id)
+        return res.send(200).send({
+            data: userDelete
+        })
+    }
+}
+
+export default UserController;
+
+// import { request } from "express"
```
Comment (MbugwaSami) [inline src/modules/users/userControllers.js]: Delete the commented code

File: src/models/user.js (hunk 2 above)
Comment (MainaMary) [reply src/models/user.js]: User data updated

File: src/modules/users/userControllers.js (hunk 4 above)
Comment (MainaMary) [reply src/modules/users/userControllers.js]: Done.

**Pull request merged**